### PR TITLE
Cray complier type detection

### PR DESCRIPTION
**Pull request by WeiqunZhang**
There are two types of Cray compilers, the classical and clang based.  The version number
is not a reliable way to determine the type.  For example, cce/9 on Cori has both
versions.  Cray compilers for aarch64 on Ookami are classical.  Thus we switch to grep
"LLVM" in `CC --version`.  This works on both Cori and Ookami.

## Checklist

The proposed changes:
- [x] fix a bug or incorrect behavior in AMReX
- [ ] add new capabilities to AMReX
- [ ] changes answers in the test suite to more than roundoff level
- [ ] are likely to significantly affect the results of downstream AMReX users
- [ ] are described in the proposed changes to the AMReX documentation, if appropriate

### CODE REVIEW

File: Tools/GNUMake/comps/cray.mak [hunk 1]
```diff
@@ -13,26 +13,18 @@ F90FLAGS =
...
-  CCE_GE_V9 := FALSE
+  CRAY_IS_CLANG_BASED = FALSE
 endif
 
 # we need to be careful on the A64fx
-ifeq ($(CRAY_CC_VERSION),10.0.1)
-  ifeq ($(CRAY_ARCH_TARGET),aarch64)
-    CCE_GE_V9 := FALSE
-  endif
+ifeq ($(CRAY_ARCH_TARGET),aarch64)
```
Comment (atmyers) [inline Tools/GNUMake/comps/cray.mak]: Why is this needed, in addition to the grep check above? Wouldn't this be false anyway on Ookami?

Comment (WeiqunZhang) [reply Tools/GNUMake/comps/cray.mak]: Oh I forgot to remove the old check.

**Review by atmyers — APPROVED** [review]
